### PR TITLE
modlib: Implement sh_addralign handling

### PR DESCRIPTION
**Pull request by yamt**
## Summary
I've seen a module with 16 bytes .rodata alignment for xmm operations.
It was getting SEGV on sim/Linux because of the alignment issue.

## Impact
CONFIG_MODULE

## Testing
The same module binary seems working fine after applying this patch.

Also, tested on sim/macOS and esp32 on qemu,
using a module with an artificially large alignment. (64 bytes)

### CODE REVIEW

File: libs/libc/modlib/modlib_load.c [hunk 1]
```diff
@@ -271,16 +293,29 @@ int modlib_load(FAR struct mod_loadinfo_s *loadinfo)
         }
     }
 #else
-  loadinfo->textalloc = (uintptr_t)lib_malloc(loadinfo->textsize +
```
Comment (xiaoxiang781216) [inline libs/libc/modlib/modlib_load.c:274]: Why go through the different patch with or without CONFIG_ARCH_USE_MODULE_TEXT?

File: libs/libc/modlib/modlib_load.c [hunk 2]
```diff
@@ -271,16 +293,29 @@ int modlib_load(FAR struct mod_loadinfo_s *loadinfo)
...
+      align = loadinfo->dataalign;
+    }
+
+  text_size = loadinfo->textsize;
+  if (loadinfo->datasize > 0)
+    {
+      text_size = _ALIGN_UP(text_size, loadinfo->dataalign);
+    }
+
+  loadinfo->textalloc = (uintptr_t)lib_memalign(align,
```
Comment (xiaoxiang781216) [inline libs/libc/modlib/modlib_load.c:308]: Is it good to combine text and data in one allocation? Since it require more large consecutive memory.

File: include/nuttx/lib/modlib.h [hunk 3]
```diff
@@ -198,6 +198,8 @@ struct mod_loadinfo_s
   uintptr_t         datastart;   /* Start of.bss/.data memory in .text allocation */
   size_t            textsize;    /* Size of the module .text memory allocation */
   size_t            datasize;    /* Size of the module .bss/.data memory allocation */
+  unsigned int      textalign;   /* Necessary alignment of .text */
+  unsigned int      dataalign;   /* Necessary alignment of .bss/.text */
```
Comment (xiaoxiang781216) [inline include/nuttx/lib/modlib.h:202]: should we use size_t like textsize/datasize?

File: libs/libc/modlib/modlib_load.c (hunk 1 above)
Comment (yamt) [reply libs/libc/modlib/modlib_load.c:274]: just because the existing code is different.

File: libs/libc/modlib/modlib_load.c (hunk 2 above)
Comment (yamt) [reply libs/libc/modlib/modlib_load.c:308]: i dunno. i guess it has both of pros and cons.

iirc, the original code was doing a single allocation and i changed it for CONFIG_ARCH_USE_MODULE_TEXT case.

File: include/nuttx/lib/modlib.h (hunk 3 above)
Comment (yamt) [reply include/nuttx/lib/modlib.h:202]: maybe. i don't think that large alignment is realistic though.

Comment (yamt) [reply include/nuttx/lib/modlib.h]: fixed

File: libs/libc/modlib/modlib_load.c (hunk 2 above)
Comment (yamt) [reply libs/libc/modlib/modlib_load.c:308]: i'm ok with making it always use separate allocations. do you want me to do it in this PR? i'm feeling that a separate PR is more appropriate.

Comment (xiaoxiang781216) [reply libs/libc/modlib/modlib_load.c:308]: Ok, it's fine to put into a new PR.

**Review by xiaoxiang781216 — APPROVED** [review]
